File: day16/day16.cpp
#include<iostream>
#include<string>
#include<unordered_map>
using namespace std;
class Solution {
    public:
    // Function is to check whether two strings are anagram of each other or not.
    bool areAnagrams(string s1, string s2) {
        unordered_map<char, int> obj;
        for (char ch : s1) {
            if (obj[ch])
                obj[ch] += 1;
            else
                obj[ch] = 1;
        }
        for (char ch : s2) {
            if (!obj[ch])
                return false;
            obj[ch] -= 1;
        }
        return true;
    }
};
int main() {
    string s1 =  "sankar", s2 = "askanr";
    cout << Solution().areAnagrams(s1, s2) << endl;
    return 0;
}

File: src/day16/day16.py
class Solution:
    #Function is to check whether two strings are anagram of each other or not.
    def areAnagrams(self, s1, s2):
        obj = {}
        for ch in s1:
            if obj.get(ch):
                obj[ch] += 1
            else:
                obj[ch] = 1
        for ch in s2:
            if not obj.get(ch):
                return False
            obj[ch] -= 1
        else:
            return True

if __name__ == '__main__':
    s1, s2 = "allergy", "allergic"
    print(Solution().areAnagrams(s1, s2))